### PR TITLE
Update Azure AD/OIDC sample

### PR DESCRIPTION
**Pull request by guardrex**
Addresses https://github.com/aspnet/Docs/issues/4449

@jmprieur Here it is. First draft, of course. :smile:

Notes:
* xplat and x-tooling:tm: I call out command-line setup vs. VS setup and config where needed.
* Derived from the `dotnet new mvc` template with `--auth SingleOrg`. They can get the sample via the command line or by cloning/downloading the repo.
* README is updated.
   * It might need a touch here and there on review, but this is a good initial draft. I'm not accustomed to working with README for built topics, so I won't be surprised if some changes are required to conform to your build system/rendered topic.
   * Guidelines adherence and style/grammar updates ... the usual stuff.
* On the mapping of Azure portal values to app config settings, I attempt to bridge the gap in technologies:
   * `Client ID` and `Application ID`
   * `Tenant ID` appearing in endpoint URLs in the portal
* I *didn't* look :eyes: to see if this sample is linked into Azure topics. If that's a potential problem (i.e., topics say things about the sample that require updates given these changes), let me know ... I'll search (tomorrow ... time for some :zzz: 😴 here lol).

cc/ @Rick-Anderson

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,70 +5,102 @@ author: jmprieur
...
-Getting started is simple!  To run this sample you will need:
-- Install .NET Core for Windows by following the instructions at [dot.net/core](https://dot.net/core), which will include Visual Studio 2015 Update 3.
-- An Internet connection
-- An Azure Active Directory (Azure AD) tenant. For more information on how to get an Azure AD tenant, please see [How to get an Azure AD tenant](https://azure.microsoft.com/en-us/documentation/articles/active-directory-howto-tenant/) 
-- A user account in your Azure AD tenant. This sample will not work with a Microsoft account, so if you signed in to the Azure portal with a Microsoft account and have never created a user account in your directory before, you need to do that now.
+### Step 1: Register the sample with your Azure Active Directory tenant
 
-### Step 1:  Clone or download this repository
+1. Sign in to the [Azure portal](https://portal.azure.com).
+2. Navigate to the Azure Active Directory blade.
```
Comment (jmprieur) [inline README.md]: We need to explain a bit more here. people might have several tenant:
2. On the top bar, click on your account and under the **Directory** list, choose the Active Directory tenant where you wish to register your application
3. Click on **More Services** in the left hand nav, and choose **Azure Active Directory**.

File: README.md [hunk 2]
```diff
@@ -5,70 +5,102 @@ author: jmprieur
...
 
-1. Open the solution in Visual Studio 2017.
-2. Open the `config.json` file.
-3. Find the `Tenant` property and replace the value with your AAD tenant name.
-4. Find the `ClientId` and replace the value with the Application ID from the Azure portal.
-5. If you changed the base URL of the sample, find the app key `ida:PostLogoutRedirectUri` and replace the value with the new base URL of the sample.
+If you aren't using [Visual Studio](https://www.visualstudio.com/vs/) as your development environment and created the project with the `dotnet new mvc` command:
 
-### Step 4:  Run the sample
+- Add Bower to the project file:
```
Comment (jmprieur) [inline README.md]: Why do we need to do that? I did not (I did Dotnet run immediately) and that worked?

File: README.md [hunk 3]
```diff
@@ -5,70 +5,102 @@ author: jmprieur
...
+  ```
 
-Clean the solution, rebuild the solution, and run it.
+### Step 3: Run the sample
 
-Click the sign-in link on the homepage of the application to sign-in.  On the Azure AD sign-in page, enter the name and password of a user account that is in your Azure AD tenant.
+Build the solution and run it.
 
-## About The Code
+Make a request to the app. The app immediately attempts to authenticate you via Azure AD. Sign in with your Global Administrator account.
```
Comment (jmprieur) [inline README.md]: Why do we need to be a global admin account? that should work with any account of the tenant?

File: Startup.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            {
+                app.UseExceptionHandler("/Home/Error");
+            }
+
+            app.UseStaticFiles();
+
+            app.UseAuthentication();
+
+            app.UseMvcWithDefaultRoute();
+        }
```
Comment (jmprieur) [inline Startup.cs]: when doing Dotnet new,  also got the following code:
           app.UseMvc(routes =>
            {
                routes.MapRoute(
                    name: "default",
                    template: "{controller=Home}/{action=Index}/{id?}");
            });

**Review by jmprieur — CHANGES_REQUESTED** [review]
Very nice, thank you.
I've proposed a few addition and have asked a question, otherwise this looks very good to me!

**Comment by guardrex** [comment]
> We need to explain a bit more here.

Yes, I was a bit overzealous in attempting to simplify the steps. I put that back. I add a link over to the tenant topic in case the reader would like more info on tenants. Also, the list there should auto-number for us with a "1." on each list item. Makes it a little easier in the future to insert and delete items.

> Why do we need to be a global admin account? that should work with any account of the tenant?

I was just trying to keep it simple. I added more information that includes using a user account.

> when doing Dotnet new, also got the following code:

Yes, they're probably going to be [switching over to the UseMvcWithDefaultRoute convenience method shortly (including for the 2.0 templates) (aspnet/templating 31)](https://github.com/aspnet/templating/issues/31). I ask them there if and how fast they plan to get a PR in, which will signal that they approve of the change. Even if they only make the change for 2.1, going ahead here with the change will harden the instructions, and a new issue won't be required to update at 2.1 (that's just a suggestion, of course). It's just one less thing to have to deal with later. :smile:

> Why do we need to do that? I did not (I did Dotnet run immediately) and that worked?

Ah ... yes. **TL;DR here** ... but there's a bit to explain on this one, so bear with me.

This is most interesting (and slightly disturbing) part! lol It turns out that this sample (and many others that were produced in VS for topics) are *broken* when running outside of VS for Bower and `BundlerMinfier.Core`.

The sample itself (the app) runs OOB (e.g., `dotnet run`) because the static assets are present in the sample ... the *wwwroot/lib* assets and minified+bundled CSS and JS are there. Bower and `BundlerMinifier.Core` features work in VS because VS has light-up features when the configuration files for these technologies are present. However, the sample is broken outside of VS for Bower and `BundlerMinfier.Core`. They simply won't run outside of VS *as there's nothing to trigger them to run*. 

As u know, there's a huge ASP.NET Core xplat push (and x-tooling for that matter, e.g., VS Code and command line). I take MS at its word on this: Samples for topics that describe and use features that work xplat and x-tooling shouldn't rely upon baked-in features of VS in order to function (logically I mean :smile:).

To make this sample run Bower and `BundlerMinifier.Core` outside of VS, the project file needs a little help. [btw- I made a small revision there on the last commit to use the `BuildBunderMinifier` package to avoid the need for the `bower install` target, which composes the project file a little better.]

Now having said all of that, they're removing Bower and `BundlerMinifier.Core` as we speak. However, they won't say what's replacing it exactly. I mean they do say that they're going to use [LibraryInstaller](https://github.com/aspnet/LibraryInstaller) (or something like it). [I've asked for more info on xplat/x-tooling scenarios (aspnet/LibraryInstaller 3)](https://github.com/aspnet/LibraryInstaller/issues/3), but they haven't revealed yet **how** they plan to make it work in **non-VS scenarios**.

What we could do here for now is follow their lead and strip out all of the Bower and `BundlerMinifier.Core` bits and leave the *wwwroot/lib* assets and minfied+bundled styles and scripts in place. The sample will run fine, as you say, but it will simply use static assets without any machinery to produce those assets. Let me know if you want to do that, and I'll make another pass to drop the tooling bits. Then when they announce what they would like to do with xplat/x-tooling samples, samples can have a pass to make them bundle, minify, and obtain static assets using LibraryInstaller.

For reference on all of that, see:
* [The template focus on VS leads to a poor non-VS dev experience (dotnet/templating 1263](https://github.com/dotnet/templating/issues/1263)
* [Remove Bower from ASP.NET Core project templates (aspnet/templating 48](https://github.com/aspnet/templating/issues/48)

**Comment by guardrex** [comment]
Now that we have their answer on the Bower/`BundlerMinifier.Core`, I removed those from the sample and from the README. We're good there. Later, they'll add Library Manager (PackMan), and the sample (and others) should get the new system for managing, minifying, and bundling static assets.

On the `UseMvcWithDefaultPath`, they still haven't decided what to do on that one yet ... there's discussion on which way they want to go. https://github.com/aspnet/templating/issues/31

I recommend leaving `UseMvcWithDefaultPath` in the sample for now. I agree with [@]Rick-Anderson on this one that using the `routes.MapRoute` is overkill and a source of confusion. If they decide to stick with the nasty one :smile:, then it's an easy/quick patch later.

**Comment by Rick-Anderson** [comment]
@jmprieur  What can I do to help get this merged? @guardrex  is under strict orders to say on vacation and not work for free. I'd like to make a few minor edits (for example, per ms style guide, avoid please), but that could be done after it's merged.

**Review by jmprieur — APPROVED** [review]

**Pull request merged**